src/components/custom-field-input-number/custom-field-input-number.test.jsx

```diff
@@ -0,0 +1,104 @@
+      const { container } = render(<TestComponent className="prioritize-me" />);
+      expect(container.firstChild).toHaveClass('prioritize-me');
+    });
+  });
+
+  describe('disabled API', () => {
+    it('can be disabled', () => {
+      const { container } = render(<TestComponent disabled />);
+      expect(container.firstChild).toHaveClass('disabled');
+      expect(screen.getByLabelText('Test label')).toBeDisabled();
+    });
+
+    it('can be enabled', () => {
+      const { container } = render(<TestComponent />);
+      expect(container.firstChild).not.toHaveClass('disabled');
+      expect(screen.getByLabelText('Test label')).not.toBeDisabled();
+    });
+  });
+
+  describe('id API', () => {
+    it('sets the id attribute', () => {
+      render(<TestComponent id="test-id" />);
+      expect(screen.getByLabelText('Test label')).toHaveAttribute('id', 'test-id');
+    });
+  });
+
+  describe('label API', () => {
+    it('sets the label', () => {
+      render(<TestComponent label="Another label" />);
+      expect(screen.getByLabelText('Another label')).toBeDefined();
+    });
+  });
+
+  describe('name API', () => {
+    it('sets the name attribute', () => {
+      render(<TestComponent name="test-name" />);
+      expect(screen.getByLabelText('Test label')).toHaveAttribute('name', 'test-name');
+    });
+  });
+
+  describe('placeholder API', () => {
+    it('sets the placeholder attribute', () => {
+      const placeholder = 'This is placeholder input';
+      render(<TestComponent placeholder={placeholder} />);
+      expect(screen.getByLabelText('Test label')).toHaveAttribute('placeholder', placeholder);
+    });
+  });
+
+  describe('required API', () => {
+    it('sets the required attribute', () => {
+      render(<TestComponent required />);
+      expect(screen.getByLabelText('Test label')).toBeRequired();
+    });
+
+    it('unsets the required attribute', () => {
+      render(<TestComponent />);
+      expect(screen.getByLabelText('Test label')).not.toBeRequired();
+    });
+  });
+
+  describe('value API', () => {
+    it('is valid on a postive integer', () => {
+      render(<TestComponent value={1} />);
+      expect(screen.getByTestId('custom-field-input')).not.toHaveClass('error');
+    });
+
+    it('is valid on zero', () => {
+      render(<TestComponent value={0} />);
+      expect(screen.getByTestId('custom-field-input')).not.toHaveClass('error');
+    });
+
+    it('is valid on a negative integer', () => {
+      render(<TestComponent value={-1} />);
+      expect(screen.getByTestId('custom-field-input')).not.toHaveClass('error');
+    });
+
+    it('is valid on a decimal integer', () => {
+      render(<TestComponent value={1.00} />);
+      expect(screen.getByTestId('custom-field-input')).not.toHaveClass('error');
+    });
+
+    it('is invalid on a decimal number', () => {
+      render(<TestComponent value={1.01} />);
+      expect(screen.getByTestId('custom-field-input')).toHaveClass('error');
+    });
```

Comment:
We should probably have some additional tests the cover incorrect usage of radix and signum. I'm also curious about how we could test the correct detection and allowance of alternate radix based on localization settings.

Reply:
I looked into this and I think we're covered by the prop-type and the number element type.

- Anything string is going to be an invalid prop-type. I am not sure about testing it since it is already expected to use the component with a real number value. 
- `--1` is invalid because it is not a real number and thus breaks in the JS token parsing (cannot test this)
- Any radix besides `.` is invalid in the number element type and it not interpreted. I think I can do a bit more googling on this but it might be best to delegate all number validations and stuff to the native browser behavior.

Reply:
I agree that delegating control of behavior to the browser makes sense... if the browser functionality is in line with our own AC. I know it wasn't ever officially a part of the AC, but it seems like if we do need internationalization in the form of locale-specific radix, then we should probably implement some sort of workaround.

Reply:
I found this snippet interesting:

> This specification does not define what user interface user agents are to use; user agent vendors are encouraged to consider what would best serve their users' needs. For example, a user agent in Persian or Arabic markets might support Persian and Arabic numeric input (converting it to the format required for submission as described above). Similarly, a user agent designed for Romans might display the value in Roman numerals rather than in decimal; or (more realistically) a user agent designed for the French market might display the value with apostrophes between thousands and commas before the decimals, and allow the user to enter a value in that manner, internally converting it to the submission format described above.

I think we're good there!

Reply:
Awesome, so it's up to the browser to implement the spec here. I wonder if Chrome does... 🤔 Also interesting was the snippet a bit lower on when it is appropriate to use a number input, especially regarding input that uses numbers but is not _actually_ a number, like credit cards.